blog/_posts/2021-12-14-sustainable-scala.md

```diff
@@ -0,0 +1,278 @@
+---
+layout: blog-detail
+post-type: blog
+by: Julien Richard-Foy, Wojciech Mazur
+title: "Sustainable Scala"
+---
+
+To what extent do programming languages affect the energy consumption of 
+programs? Where does Scala stand in that picture? Does that matter for 
+building an environmentally sustainable digital world?
+
+We found out that the most important lever is to **extend the lifetime of 
+hardware**. Nevertheless, the energy consumption due to **running software** is 
+important, and it **varies significantly between programming languages**.
+Scala is well positioned for a high-level language, and **writing low-level 
+imperative code** can make a difference.
+
+## The world is warming, what about the digital world?
+
+According to [climate experts], it is urgent to assess the environmental 
+impact of our decisions (in all areas) to significantly reduce the 
+greenhouse-gas (GHG) emissions of human activities.
+
+In 2019, the sector of IT was responsible for 3.5% of the worldwide GHG 
+emissions [^1] [^2], which is similar to the aviation sector. More 
+importantly, this number is growing exponentially, and it is expected to 
+double between now and 2025 [^1]. This trajectory is obviously not 
+sustainable. The GHG emissions of the IT sector must decrease.
+
+In that regard, good news is that more and more IT companies are committing to 
+targeting carbon neutrality, including Scala companies such as [Netflix], 
+[Zalando], and [Stripe].
+
+Where do we start? If we break down the GHG emissions within the IT sector, 
+we observe that most of them happen during the manufacturing process of 
+hardware rather than during the usage of hardware [^1] [^3]. Therefore, the 
+first action point is to **extend the lifetime of hardware**.
+
+Then, we can have a look at the emissions due to using hardware, namely
+the emissions due to running software.
+
+## Do programming languages affect energy consumption?
+
+The GHG emissions due to running software are caused by the energy 
+consumption of computers. Obviously, the more computations are performed, the 
+more energy is consumed. Said otherwise, the nature of the program (whether 
+it is computationally intensive or not) is the most important factor.
```

Comment:
compare to what? 

the time of the execution, the network or disk throughput maybe?

Reply:
What I meant is that a program that computes a lot of things is likely to consume more energy than a program that just adds two numbers, even if the latter is written in an inefficient language.

Reply:
Okay so maybe you can make it clear with something like:

```
Said otherwise, the nature of the program is more important than the language in which it is written.

Reply:
Thank you for the suggestion, fixed in f55c814